src/Topology.tsx

```diff
@@ -646,19 +639,19 @@ export class Topology extends React.Component<Props, {}> {
 
         var children = new Array<NodeWrapper>()
         node.children.forEach(child => {
-            var field = child.wrapped.data.Type
-            if (!field) {
+            var nodeType = child.wrapped.data.Type
+            if (!nodeType) {
                 children.push(child)
                 return
             }
 
-            var gid = node.id + "_" + field + "_" + child.wrapped.getWeight()
+            var gid = this.props.groupGID!(node.wrapped, child.wrapped)
             if (pushed.has(gid)) {
                 return
             }
 
             var wrapper = groups.get(gid)
-            if (wrapper && wrapper.wrapped.children.length > 5) {
```

Comment:
you should remove this condition otherwise we have group for all the node, meaning group on 1 node example, I suggest to fix this like this:

```
 var groupSize = this.props.groupSize || defaultGroupSize
 var wrapper = groups.get(gid)

 if (wrapper && wrapper.wrapped.children.length > groupSize) {
```

you can use the local groupSize line 668